testsuite/tests/basic/switch_opts.ml

```diff
@@ -0,0 +1,62 @@
+(* Test for optimisation of jump tables to arrays of constants *)
+
+let p = Printf.printf
+
+type test =
+  Test : 'b * 'a * ('b -> 'a) -> test
+
+type t = A | B | C
+
+(* These test functions need to have at least three cases.
+   Functions with fewer cases don't trigger the optimisation,
+   as they are compiled to if-then-else, not switch *)
+let testcases = [
+  Test (3, 3, function 1 -> 1 | 2 -> 2 | 3 -> 3 | _ -> 0);
+  Test (3, -3, function 1 -> 1 | 2 -> 2 | 3 -> -3 | _ -> 0);
+  Test (3, min_int, function 1 -> 1 | 2 -> 2 | 3 -> min_int | _ -> 0);
+  Test (3, max_int, function 1 -> 1 | 2 -> 2 | 3 -> max_int | _ -> 0);
+  Test (3, 3., function 1 -> 1. | 2 -> 2. | 3 -> 3. | _ -> 0.);
+  Test (3, ""^"c"^"", function 1 -> "a" | 2 -> "b" | 3 -> "c" | _ -> "");
```

Comment:
It's not terribly important. I was imagining a bizarre failure mode where the table is generated but the constants are screwed up, and the test happens to pass because it doesn't actually examine the constants because they're the same object.

I've changed it to use `Sys.opaque_identity`